### PR TITLE
Feature/allied gs950

### PR DESCRIPTION
**Pull request by julsemaan**
# Description
Adds support for Allied Telesis GS950

# Impacts
FreeRADIUS policies

# Delete branch after merge
YES

# NEWS file entries
## New Features
* Added support for Allied Telesis GS950 switches

### CODE REVIEW

File: docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc [hunk 1]
```diff
@@ -654,6 +654,79 @@ Finally, enable the necessary 802.1X and VOIP settings on each interface:
     lldp med network-policy add 1
     lldp med network-policy add 2
 
+GS950
+^^^^^
+
+PacketFence supports the GS950 switch using :
+
+* MAC Authentication (mac-only)
```
Comment (extrafu) [inline docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc]: I don't like "mac-only" - remove it or clarify it

File: docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc [hunk 2]
```diff
@@ -654,6 +654,79 @@ Finally, enable the necessary 802.1X and VOIP settings on each interface:
...
+* 802.1X (without fallback to MAC authentication)
+
+Global configuration
+++++++++++++++++++++
+
+First, ensure that the VLANs you want to assign are part of the VLAN database via the following page:
+
+Note that they only need to be tagged on the trunk and don't need any specific configuration for the dynamic VLAN assignment here.
```
Comment (extrafu) [inline docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc]: Move that below the image?

File: docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc [hunk 3]
```diff
@@ -654,6 +654,79 @@ Finally, enable the necessary 802.1X and VOIP settings on each interface:
...
+Ensure you configure at least:
+
+[options="compact"]
+* Type: Allied Telesis GS950
+* RADIUS secret: useStrongerSecret
+* SNMP Version: v2c
+* SNMP Community Read: private
+* SNMP Community Write: private
+
+If you are using MAC authentication on this switch, tyou must adjust the FreeRADIUS configuration so it transforms the EAP requests this switch sends into requests that PacketFence will interpret as MAC authentication. This is necessary because the 'MAC Based' mode of the switch still performs EAP requests.
```
Comment (extrafu) [inline docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc]: tyou!

Comment (julsemaan) [reply docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc]: That's in the new updated English. It is a formal 'you'

Note that this doc was approved by @TheGreatZayme

File: lib/pf/Switch/AlliedTelesis/GS950.pm [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+
+=back
+
+=head1 AUTHOR
+
+Inverse inc. <info@inverse.ca>
+
+=head1 COPYRIGHT
+
+Copyright (C) 2005-2016 Inverse inc.
```
Comment (fdurand) [inline lib/pf/Switch/AlliedTelesis/GS950.pm]: 2017

**Comment by dwlfrth** [comment]
@julsemaan rebase

File: docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc [hunk 5]
```diff
@@ -654,6 +654,79 @@ Finally, enable the necessary 802.1X and VOIP settings on each interface:
...
+
+image::docs/images/allied-telesis-gs950/port-control-MAB.png[scaledwidth="100%",alt="MAB config"]
+
+802.1x
+++++++
+
+Go in Security->Port Access Control, select the port you want to enable MAB on, and ensure you set:
+
+[options="compact"]
+* Authentication Mode: MAC Based
```
Comment (fdurand) [inline docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc]: Authentication Mode: 802.1X

File: docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc [hunk 6]
```diff
@@ -654,6 +654,79 @@ Finally, enable the necessary 802.1X and VOIP settings on each interface:
...
+* Port Control: Auto
+* Supplicant Mode: Single
```
Comment (fdurand) [inline docs/PacketFence_Network_Devices_Configuration_Guide.asciidoc]: Supplicant Mode: Multiple

**Review by fdurand — COMMENTED** [review]
Fix the documentation and i will merge

**Comment by julsemaan** [comment]
@fdurand 

Touchups on doc done

+Thanks for fixing the merge conflict, @lzammit will pay you a beer

**Pull request merged**